orchestration/_tests/test_globus_flow.py

```diff
@@ -0,0 +1,421 @@
+
+        pruning_config = JSON(value={"max_wait_seconds": 600})
+        pruning_config.save(name="pruning-config")
+        yield
+
+
+@pytest.fixture(autouse=True)
+def set_env_vars(monkeypatch):
+    """
+    A pytest fixture that automatically sets environment variables required for
+    Globus Compute SDK and Globus Flows during each test.
+
+    The fixture uses monkeypatching to mock environment variables, ensuring
+    that the tests don't rely on real environment configurations.
+
+    Args:
+        monkeypatch: A pytest fixture that allows modifying environment variables
+                     for the duration of the test.
+
+    Yields:
+        None
+    """
+    monkeypatch.setenv("GLOBUS_COMPUTE_ENDPOINT", uuid4())
+    monkeypatch.setenv("GLOBUS_RECONSTRUCTION_FUNC", uuid4())
+    monkeypatch.setenv("GLOBUS_IRIBETA_CGS_ENDPOINT", uuid4())
+
+
+@pytest.fixture(autouse=True)
+def mock_globus_compute_client(monkeypatch):
+    """
+    A pytest fixture that automatically mocks the Globus Compute SDK's Client
+    class to avoid real network calls during testing.
+
+    This fixture replaces the Client class methods with mock implementations
+    that return predefined values, ensuring that tests are isolated from
+    external dependencies.
+
+    Args:
+        monkeypatch: A pytest fixture that allows monkeypatching class methods
+                     for the duration of the test.
+
+    Yields:
+        None
+    """
+    monkeypatch.setattr(Client, "__init__", MockGlobusComputeClient.__init__)
+    monkeypatch.setattr(Client, "version_check", MockGlobusComputeClient.version_check)
+    monkeypatch.setattr(Client, "run", MockGlobusComputeClient.run)
+    monkeypatch.setattr(Client, "get_task", MockGlobusComputeClient.get_task)
+    monkeypatch.setattr(Client, "get_result", MockGlobusComputeClient.get_result)
+
+
+@pytest.fixture
+def mock_confidential_client():
+    """Fixture for mocking the confidential client"""
+    with patch('orchestration.globus.flows.ConfidentialAppAuthClient') as MockClient:
+        instance = MockClient.return_value
+        instance.oauth2_client_credentials_tokens.return_value = {
+            "access_token": "fake_access_token",
+            "expires_in": 3600
+        }
+        yield instance
+
+
+@pytest.fixture
+def mock_specific_flow_client():
+    """Fixture for mocking the specific flow client"""
+    with patch('orchestration.globus.flows.SpecificFlowClient') as MockClient:
+        instance = MockClient.return_value
+        instance.scopes = MagicMock()
+        instance.scopes.make_mutable.return_value = "fake_scope"
+        yield instance
+
+
+@pytest.fixture
+def mock_client_credentials_authorizer():
+    """Fixture for mocking the client credentials authorizer"""
+    with patch('orchestration.globus.flows.ClientCredentialsAuthorizer') as MockAuthorizer:
+        instance = MockAuthorizer.return_value
+        yield instance
+
+
+class FlowDefinition(BaseModel):
```

Comment:
It would be interseting to see these models used in the actual code rather than just the tests. Perhaps a future PR?